### DISCUSSION TITLE
How to link locally built third party libraries

### DISCUSSION THREAD

**Discussion by Jim-215-Fisher** [fortran-lang/fpm#477] [General]
I was trying to use fpm to build a Fortran project. The project requires several locally built libraries. How do I specify the paths and libraries as well as mod files in fpm.toml?

**Comment by milancurcic** [comment]
Hi @Jim-215-Fisher,

See [this section](https://github.com/fortran-lang/fpm/blob/master/manifest-reference.md#link-external-libraries) of the manifest reference for linking.

To use external mod files (that is, not the ones from fpm package dependencies), see [this section](https://github.com/fortran-lang/fpm/blob/master/manifest-reference.md#use-system-installed-modules).

fpm will still need to know where to find your external mod files, and there are two ways to do it:

1. Copy or symlink them into the `include/` directory in the root of your fpm project. This directory is added to the include and modules path by default.
2. Specify a path on the command line, i.e. `fpm build --flag '-I/path/to/mod/files'`.
